### PR TITLE
Moar fetch

### PR DESCRIPTION
**Pull request by gregtalarico**

### CODE REVIEW

File: src/utils/fetch-utils.js [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
-                        }
-                    })
-                    .catch((e)=>reject(e)); //error processing json response      
+            .then((res) => {
+                if (res.status === 204){
+                    resolve();
+                } else {
+                    res.json()
+                        .then((json)=>{
+                            if(res.ok){ //200 level error
```
Comment (benschiefer) [inline src/utils/fetch-utils.js]: //200 level response

File: src/utils/fetch-utils.js [hunk 2]
```diff
@@ -1,25 +1,39 @@
...
+            .catch((e)=>reject(e)); //fetch error
     })
     
     return plomise;
-};
\ No newline at end of file
+};
+
+
+export const doGet = (url, body) => doFetch(url, 'POST', body)
```
Comment (benschiefer) [inline src/utils/fetch-utils.js]: whats the utility of this?

**Review by benschiefer — CHANGES_REQUESTED** [review]

File: src/utils/fetch-utils.js (hunk 2 above)
Comment (gregtalarico) [reply src/utils/fetch-utils.js]: mimic a 404 response one would expect from a GET if the item does not exist

File: src/utils/fetch-utils.js (hunk 1 above)
Comment (gregtalarico) [reply src/utils/fetch-utils.js]: fine

File: src/utils/fetch-utils.js (hunk 2 above)
Comment (gregtalarico) [reply src/utils/fetch-utils.js]: also resolve with the first item in the search results array, instead of having to do that over and over

**Review by benschiefer — APPROVED** [review]

**Pull request merged**